Mulligan Rounds for Wizard/Blob/Malf

Incoming5643 opened the pull request:
On primary antagonist rounds (wizard/blob/malfunction) if the config is set for continuous rounds the round will convert on the fly to a new mode chosen from these modes if the primary antagonist dies:

EXTENDED
TRAITOR
CHANGELING
DOUBLE AGENTS
TRAITORLING

based on the game probabilities set through the config.

The round will not convert if more than 30% of the players are dead (round immediately ends) or if the shuttle is beyond the point of no return (the round ends normally).

Adds the CONTINUOUS_ROUND_BLOB config option

This pull is made in response to this forum topic (https://tgstation13.org/phpBB/viewtopic.php?f=2&t=2966) and if any continuous configs are turned on I strongly recommend rebalancing the probabilities of wizard/blob/malfunction to account for the fact that said modes can be run more often without many of the gripes brought up in the thread.

Magic Mania 1.861 | Blob Mania 1.1 | Malfunction Mania 0.1

Fixes #2679

---

Adds a new toggle: Midround_antagonist that allows players to opt out of gaining antagonist through this system if they'd prefer. It defaults to on but is newbie safe because without any antagonist flags (which default to off), it won't place players into antagonist slots.

Incoming5643 committed:
On primary antagonist rounds (wizard/blob/malfunction) if the config is set for continous rounds the round will convert on the fly to a new mode choosen from these mode if the primary antagonist dies:

EXTENDED
TRAITOR
CHANGELING
DOUBLE AGENTS
TRAITORLING

based on the game probabilities set through the config.

The round will not convert if more than 30% of the players are dead (round immediatly ends) or if the shuttle is beyond the point of no return (the round ends normally).

Adds the CONTINIOUS_ROUND_BLOB config

Incoming5643 committed:
adds changelog

Iamgoofball commented:
>extended

plsno

paprka commented:
Don't add extended to this, if we're gonna have continuous rounds we need a driving force towards the round ending because once these antags die there is pretty much no wiggle room in regards to more fun unless admins do shit (lol)

Incoming5643 commented:
It runs off the Game probabilities, it will never be extended unless extended was a possible option for the round to begin with. On sybil mulligan would convert to extended a whopping 4% of the time

paprka commented:
Ah, thanks for the clarification. That's fine then.

Iamgoofball commented:
oh okay makes sense

Cheridan commented:
Extended is supposed to be for having slow RP rounds or for admins to goof off on, in either case having an antag pop in out of nowhere and ending the round could be pretty disruptive. Does the round end immediately if it's Extended -> Wizard and the wiz dies?

Tentative :+1: but we gotta make sure to account for everything when it comes to gamemode stuff like this. :smiley:

Incoming5643 commented:
These rounds convert from wizard/blob/malf to extended/traitor/traitorling/ling/DA, never the other way around, so the sanctity of extended is safe, this isn't the tensioner.

MrStonedOne commented on code/game/gamemodes/game_mode.dm:
```diff
@@ -88,6 +90,50 @@
+	for(var/mob in player_list)
+		if(mob in living_mob_list)
+			living_crew += mob
+	if(living_crew.len / player_list.len <= 0.7) //If a lot of the player base died, we start fresh
```
Wouldn't this also count "observe" ghosts and admin ghosts as dead players, or are they not included in player_list? And what about braindeads, (such as admin ghosted peoples)?

What exactly is included in player_list i guess? Because something like this should exclusively only count people on the manifest.

Incoming5643 replied:
It's not the best measure no, but it's surprisingly hard to quantify "players that matter".

Ikarrus commented:
Joined_player_list

Pennwick commented:
I had a thought on this earlier. Would it be possible to have a brief window of time where players could respawn after the Primary Antag goes down? This way people could get back in the round even if their body was destroyed or lost.

Dorsisdwarf commented:
Nah

Profakos commented:
Could there be a preference  option for becoming a mulligan antag? Personally I wouldn't mind suddenly becoming one, but some of my friends would find it a jarring.

Incoming5643 commented:
Currently antag flags are Boolean, they could be modified to be "Yes" "Roundstart Only" "No", but that's another pull.

RemieRichards commented:
Just add a "be Mulligan Antag" or something more descriptive.

Ikarrus commented:
As much as I love this idea, I'm one of those who would hate being converted to an antaginist mid round. I find it really disruptive to my experience. So yeah, a preference option as RR suggested would be very much appreciated.

Incoming5643 committed:
Makes the "are too many people dead for this to be fun?" check more indicative of people who are actually playing

Adds a new toggle: Midround_antagonist that allows players to opt out of gaining antagonist through this system if they'd prefer. It defaults to on but is newbie safe because without any antagonist flags (which default to off), it won't place players into antagonist slots.

Incoming5643 commented:
It's done, fixed up the living player check too

Kromgar commented:
Ikarrus you were a sleeper agent all along. Thats a good idea... after the antag for those rounds is dead have a mysterious message be announced via centcomm report kinda thing saying 

Codeword: X activating sleeper agents as a ic reason for it

Ikarrus commented:
I don't care what the fluff is, I don't want to play as an antagonist once I've already started playing.

Incoming5643 commented:
don't talk him out of it, the fewer people in the draft the better chance you have of getting it yourself :trophy:
